### DISCUSSION TITLE
Cursor moves randomly when formatting with prettier

### DISCUSSION THREAD

**Discussion by cdaein** [LazyVim/LazyVim#1354] [Q&A]
Hi,

And I am having issue with cursor moving randomly when formatting with prettier in TypeScript. It happends to relatively longer lines.

Has anyone experienced this issue?

I have a fresh LazyVim install with prettier added to `require('lazy').setup({ spec })` with the builtin extra plugin:
```
 { import = "lazyvim.plugins.extras.formatting.prettier" },
```

I have also tried to install prettier and prettierd (not using builtin import) but they all seem to have the same issue.

https://github.com/LazyVim/LazyVim/assets/3522003/5291cafb-6101-40b0-863d-adb49a1c7d95

**Comment by timjackleus** [comment]
Hi, I've never experienced this issue myself and all I've done is followed the recipe from the website like you have, and installed prettierd globally
```
npm install -g @fsouza/prettierd
```
Did you add [this](https://www.lazyvim.org/extras/formatting/prettier#null-lsnvim) to your config as well?

**Reply by cdaein** [reply]
thank you for the suggestion. I have followed the config as the link you mentioned, but same issue.
i also added NPM package like you have (which I didn't before) but it's the same.

My neovim version is:
```
NVIM v0.9.1
Build type: Release
LuaJIT 2.1.0-beta3
```

**Reply by yushaku** [reply]
Did you use LSP-sage? i have the same problem as you when I add lsp-sage to my Nvim

**Comment by bwafi** [comment]
have found a solution? i have the same problem

**Comment by abeldekat** [comment]
Hello @timjackleus @bwafi @yushaku 

Although I don't use typescript myself, this topic caught my attention.
Looking at your personal repositories, file `lua/config/lazy.lua`: Maybe we can rule out `mini.animate` as a factor.

@bwafi @yushaku 
Could you `disable` the line containing `mini.animate` in your `lazy.lua`?

@timjackleus 
Could you `enable` aforementioned line?

**Reply by abeldekat** [reply]
I am trying to reproduce the issue, without success.
Plugin mini.animate can be ruled out, though.

I am using a new LazyVim starter installation, adding one line to lua/config/lazy.lua:
`{ import = "lazyvim.plugins.extras.formatting.prettier" }`,

The version of prettierd, installed by mason, is 0.24.2

**Reply by timjackleus** [reply]
Hi @abeldekat I tried to enable mini.animate but sadly still unable to reproduce the problem.
